phpdotnet/phd/Package/PHP/Web.php

```diff
@@ -211,6 +223,40 @@ protected function writeJsonIndex() {
         v("Index written", VERBOSE_FORMAT_RENDERING);
     }
 
+    protected static function generateDeprecatedInfo($filename) {
+        static $info;
```

Comment:
After reconsidering what you said, I have mixed feelings. `generateVersionInfo()` should return array unless we don't want to insert deprecated deprecation state directly into `$this->versions`, making it multidimensional array.

It would also require second condition, either way, because when generating indices for `$this->depracted` I need to replace special characters with underscores, not hyphens. Otherwise, I wouldn't be able to compare them with result of `Format::getShortDescription()`.

```php
// generateVersionInfo()
$funcname = str_replace(
array("::", "->", "__", "_", '$'),
array("-",  "-",  "-",  "-", ""),
$r->value)

// generateDeprecatedInfo()
$funcname = str_replace(
array("::", "->", "__", "_", '$'),
array("_",  "_",  "_",  "_", ""),
$r->value)
```

What do you think?